src/components/CurrentScenarioSelector/CurrentScenarioSelector.js

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { HierarchicalComboBox } from '@cosmotech/ui';
+import { useSortedScenarioList } from '../../hooks/ScenarioListHooks';
+import { STATUSES } from '../../state/commons/Constants';
+import { useCurrentScenarioData, useCurrentScenarioStatus, useFindScenarioById } from '../../state/hooks/ScenarioHooks';
+import { t } from 'i18next';
+
+const CurrentScenarioSelector = ({ disabled, renderInputToolType }) => {
+  const sortedScenarioList = useSortedScenarioList();
+  const currentScenarioData = useCurrentScenarioData();
+  const currentScenarioStatus = useCurrentScenarioStatus();
+
+  const changeScenario = useFindScenarioById();
+
+  const handleScenarioChange = useCallback(
+    (event, scenario) => {
+      changeScenario(scenario.id);
+    },
+    [changeScenario]
+  );
+
+  const noScenario = currentScenarioData === null;
+  const scenarioListDisabled = sortedScenarioList.length === 0 || noScenario;
```

Comment:
We could probably improve this part by writing instead:
`const noScenario = currentScenarioData === null || sortedScenarioList.length === 0;`

and using it in the props of the returned componet as:
`disabled={disabled || noScenario || isSwitchingScenario}`

Reply:
done